cdk/lib/recipes-reindex.ts

```diff
@@ -0,0 +1,182 @@
+import type { GuStack } from '@guardian/cdk/lib/constructs/core';
+import { GuLambdaFunction } from '@guardian/cdk/lib/constructs/lambda';
+import { Duration, RemovalPolicy } from 'aws-cdk-lib';
+import { Effect, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Architecture, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import {
+	Choice,
+	Condition,
+	CustomState,
+	Fail,
+	StateMachine,
+	Succeed,
+	TaskInput,
+	Wait,
+	WaitTime,
+} from 'aws-cdk-lib/aws-stepfunctions';
+import { LambdaInvoke } from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { Construct } from 'constructs';
+
+type RecipesReindexProps = {
+	contentUrlBase: string;
+	reindexBatchSize: string;
+};
+
+export class RecipesReindex extends Construct {
+	constructor(
+		scope: GuStack,
+		id: string,
+		{ contentUrlBase, reindexBatchSize }: RecipesReindexProps,
+	) {
+		super(scope, id);
+
+		const appBase = 'recipes-reindex';
+		const lambdaFileName = `${appBase}.zip`;
+
+		const snapshotBucket = new Bucket(this, 'staticServing', {
```

Comment:
little niggle - can we get a better name than `staticServing`? Something like `reindexSnapshots`?

Reply:
Ctrl-C snafu. aeee945ae5fe9728a77db534844b7eafbb733ac0 👍